ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/Placeholder.kt

```diff
@@ -61,112 +53,163 @@ internal object PlaceholderDefaults {
+)
 
-    // The current highlight animation progress
-    var highlightProgress: Float by remember { mutableStateOf(0f) }
+// The private class that implements the placeholder logic as a Modifier.Node
 
-    // This is our crossfade transition
-    val transitionState = remember { MutableTransitionState(visible) }.apply {
-        targetState = visible
-    }
-    val transition = updateTransition(transitionState, "placeholder_crossfade")
+@Suppress("LongParameterList", "MagicNumber")
+private class PlaceholderNode(
+    var visible: Boolean,
+    var color: Color,
+    var shape: Shape,
+    var highlight: PlaceholderHighlight?,
+    var placeholderFadeTransitionSpec: () -> FiniteAnimationSpec<Float>,
+    var contentFadeTransitionSpec: () -> FiniteAnimationSpec<Float>,
+) : Modifier.Node(), DrawModifierNode {
 
-    val placeholderAlpha by transition.animateFloat(
-        transitionSpec = placeholderFadeTransitionSpec,
-        label = "placeholder_fade",
-        targetValueByState = { placeholderVisible -> if (placeholderVisible) 1f else 0f },
-    )
-    val contentAlpha by transition.animateFloat(
-        transitionSpec = contentFadeTransitionSpec,
-        label = "content_fade",
-        targetValueByState = { placeholderVisible -> if (placeholderVisible) 0f else 1f },
-    )
+    private var lastSize: Size? = null
+    private var lastLayoutDirection: LayoutDirection? = null
+    private var lastOutline: Outline? = null
 
-    // Run the optional animation spec and update the progress if the placeholder is visible
-    val animationSpec = highlight?.animationSpec
-    if (animationSpec != null && (visible || placeholderAlpha >= 0.01f)) {
-        val infiniteTransition = rememberInfiniteTransition(label = "Transition")
-        highlightProgress = infiniteTransition.animateFloat(
-            initialValue = 0f,
-            targetValue = 1f,
-            animationSpec = animationSpec,
-            label = "Animation",
-        ).value
-    }
+    private val placeholderAlpha = Animatable(if (visible) 1f else 0f)
+    private val contentAlpha = Animatable(if (visible) 0f else 1f)
+    private val highlightProgress = Animatable(0f)
+    private val paint = Paint()
+
+    private var highlightAnimationJob: Job? = null
 
-    val paint = remember { Paint() }
-    remember(color, shape, highlight) {
-        drawWithContent {
-            // Draw the composable content first
-            if (contentAlpha in 0.01f..0.99f) {
-                // If the content alpha is between 1% and 99%, draw it in a layer with
-                // the alpha applied
-                paint.alpha = contentAlpha
-                withLayer(paint) {
-                    with(this@drawWithContent) {
-                        drawContent()
+    // onAttach is called when the node is attached to the composition.
+    override fun onAttach() {
+        // Coroutine for the crossfade animation
+        coroutineScope.launch {
+            snapshotFlow { visible }
```

Comment:
`isVisible` is a regular variable, so `snapshotFlow { isVisible }` will not observe updates to it 🙏